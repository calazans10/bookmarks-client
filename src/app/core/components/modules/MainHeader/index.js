import React from 'react';
import PropTypes from 'prop-types';
import { Container, Wrapper } from './style';

const MainHeader = ({ children }) => (
  <Container>
    <Wrapper>{children}</Wrapper>
  </Container>
);

MainHeader.propTypes = {
  children: PropTypes.node.isRequired,
};

export default MainHeader;
