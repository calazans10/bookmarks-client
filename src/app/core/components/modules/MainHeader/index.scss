@import 'styles/utilities';

.main-header {
  padding: rem(20);
  border-bottom: rem(1) solid #efefef;
  background-color: #f8f8f8;

  a {
    color: #7d7c7a;

    &:focus,
    &:hover {
      color: #7d7c7a;
    }
  }

  &__container {
    display: flex;
    align-content: center;

    @media(min-width: 1044px) {
      max-width: rem(1044);
      margin-right: auto;
      margin-left: auto;
    }
  }
}
