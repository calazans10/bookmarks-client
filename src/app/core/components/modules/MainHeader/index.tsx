import React from 'react';
import { Container, Wrapper } from './style';

type MainHeaderProps = {
  children: React.ReactNode;
};

const MainHeader = ({ children }: MainHeaderProps) => (
  <Container>
    <Wrapper>{children}</Wrapper>
  </Container>
);

export default MainHeader;
