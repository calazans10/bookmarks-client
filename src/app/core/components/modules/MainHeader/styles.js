import styled from 'styled-components';
import { rem } from 'polished';

export const Container = styled.header`
  padding: ${rem('20px')};
  border-bottom: ${rem('1px')} solid #efefef;
  background-color: #f8f8f8;

  a {
    color: #7d7c7a;

    &:focus,
    &:hover {
      color: #7d7c7a;
    }
  }
`;

export const Wrapper = styled.div`
  display: flex;
  align-content: center;

  @media (min-width: 1044px) {
    max-width: ${rem('1044px')};
    margin-right: auto;
    margin-left: auto;
  }
`;
