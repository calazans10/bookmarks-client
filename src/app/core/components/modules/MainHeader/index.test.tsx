import React from 'react';
import { render } from '@testing-library/react';
import MainHeader from './';

describe('MainHeader', () => {
  it('renders without crashing', () => {
    const { container } = render(
      <MainHeader>
        <ul>
          <li>Test</li>
        </ul>
      </MainHeader>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
