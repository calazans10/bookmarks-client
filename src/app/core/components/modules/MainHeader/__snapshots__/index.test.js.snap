// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainHeader renders without crashing 1`] = `
<header
  class="sc-AykKC iYFxAq"
>
  <div
    class="sc-AykKD enPysH"
  >
    <ul>
      <li>
        Test
      </li>
    </ul>
  </div>
</header>
`;
